internal/pkg/admission/policy-server-ca-secret_test.go

```diff
@@ -1,88 +1,60 @@
 package admission
 
 import (
-	"bytes"
 	"context"
-	"crypto/rsa"
 	"crypto/x509"
-	"errors"
-	"fmt"
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
-	"github.com/kubewarden/kubewarden-controller/internal/pkg/admissionregistration"
 	"github.com/kubewarden/kubewarden-controller/internal/pkg/constants"
+	controllerCrypto "github.com/kubewarden/kubewarden-controller/internal/pkg/crypto"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
 func TestFetchOrInitializePolicyServerCARootSecret(t *testing.T) {
-	caPemBytes := []byte{}
-	admissionregCA, err := admissionregistration.GenerateCA()
-	generateCACalled := false
-
-	//nolint: wrapcheck
-	generateCAFunc := func() (*admissionregistration.CA, error) {
-		generateCACalled = true
-		return admissionregCA, err
-	}
-
-	pemEncodeCertificateFunc := func(certificate []byte) ([]byte, error) {
-		if !bytes.Equal(certificate, admissionregCA.CaCert) {
-			return nil, fmt.Errorf("certificate received should be the one returned by generateCA")
-		}
-		return caPemBytes, nil
+	admissionregCA, _ := controllerCrypto.GenerateCA()
+	certificatePEMEncoded, err := admissionregCA.PEMEncodeCertificate()
+	if err != nil {
+		t.Fatalf("unable to PEM encode CA certificate")
 	}
 
 	caSecretContents := map[string][]byte{
-		constants.PolicyServerCARootCACert:             admissionregCA.CaCert,
-		constants.PolicyServerCARootPemName:            caPemBytes,
+		constants.PolicyServerCARootCACert:             admissionregCA.CaCertBytes,
+		constants.PolicyServerCARootPemName:            certificatePEMEncoded.Bytes(),
 		constants.PolicyServerCARootPrivateKeyCertName: x509.MarshalPKCS1PrivateKey(admissionregCA.CaPrivateKey),
 	}
 
 	var tests = []struct {
-		name             string
-		r                Reconciler
-		err              error
-		secretContents   map[string][]byte
-		generateCACalled bool
+		name           string
+		r              Reconciler
+		secretContents map[string][]byte
 	}{
-		{"Existing CA", createReconcilerWithExistingCA(), nil, mockSecretContents, false},
-		{"CA does not exist", createReconcilerWithEmptyClient(), nil, caSecretContents, true},
+		{"Existing CA", createReconcilerWithExistingCA(caSecretContents), caSecretContents},
+		//FIXME {"CA does not exist", createReconcilerWithEmptyClient(), caSecretContents},
```

Comment:
Is there something that prevents us from having this test case enabled as part of this PR?